admin.go

```diff
@@ -373,18 +412,35 @@ func (ca *clusterAdmin) CreatePartitions(topic string, count int32, assignment [
 		return err
 	}
 
-	rsp, err := b.CreatePartitions(request)
-	if err != nil {
-		return err
-	}
+	for i := 0; i < ca.conf.Admin.Retry.Max; i++ {
+		rsp, err := b.CreatePartitions(request)
+		if err != nil {
+			return err
+		}
 
-	topicErr, ok := rsp.TopicPartitionErrors[topic]
-	if !ok {
-		return ErrIncompleteResponse
-	}
+		topicErr, ok := rsp.TopicPartitionErrors[topic]
+		if !ok {
+			return ErrIncompleteResponse
+		}
+
+		if topicErr.Err != ErrNoError {
+			if topicErr.Err == ErrNotController {
+				b, err = ca.refreshController()
+				if err != nil {
+					// if unable to refresh controller just return original err
+					return topicErr
+				}
+				// retriable error
+				time.Sleep(ca.conf.Admin.Retry.Backoff)
+				Logger.Printf(
+					"admin/createpartitions retrying after %dms... (%d attempts remaining)\n",
+					ca.conf.Admin.Retry.Backoff/time.Millisecond, ca.conf.Admin.Retry.Max-i)
+				continue
+			}
```

Comment:
wondering if these duplicated codes could be removed and have only a single function to have this specific error handling?

looks the difference is the methods returning different error types and the logged message.

Wondering if some interface/struft could be created to make `DeleteTopicsResponse` return something that has an error with the form of:
```go
type Xxx struct {
	Err    KError
	ErrMsg *string
}
```

similar to the other error types:
``` go
type CreateTopicsResponse struct {
	Version      int16
	ThrottleTime time.Duration
	TopicErrors  map[string]*TopicError
}
type TopicError struct {
	Err    KError
	ErrMsg *string
}

type CreatePartitionsResponse struct {
	ThrottleTime         time.Duration
	TopicPartitionErrors map[string]*TopicPartitionError
}

type TopicPartitionError struct {
	Err    KError
	ErrMsg *string
}

type CreateTopicsResponse struct {
	Version      int16
	ThrottleTime time.Duration
	TopicErrors  map[string]*TopicError
}

type DeleteTopicsResponse struct {
	Version         int16
	ThrottleTime    time.Duration
	TopicErrorCodes map[string]KError
}

KError
```

Reply:
Yeah I'd like to fix that discrepancy in the return type error code, but wasn't sure if that would need a 2.x bump as a change in api

Reply:
yeah right, that'll be a breaking change, so we need to wait for 2.x